crates/test-utils/src/deploy_helpers.rs

```diff
@@ -831,16 +742,81 @@ impl KakarotTestEnvironmentContext {
         Self { sequencer, kakarot_client, kakarot, kakarot_contract, evm_contracts }
     }
 
-    pub async fn deploy_evm_contract<T: Tokenize>(mut self, contract_args: ContractDeploymentArgs<T>) -> Self {
-        let kakarot = &self.kakarot;
-        let sequencer = &self.sequencer;
-        let evm_contracts = &mut self.evm_contracts;
+    /// Initializes the Kakarot test environment and deploys the Kakarot system and EVM contracts.
+    async fn initialize_without_dumped_state(
+        sequencer: TestSequencer,
+        starknet_provider: Arc<JsonRpcClient<HttpTransport>>,
+    ) -> Self {
+        let starknet_account = sequencer.account();
+
+        // Define the funding amount for the Kakarot system
+        let funding_amount = FieldElement::from(1000000000000000000_u64);
+
+        // Deploy the Kakarot system
+        let kakarot = deploy_kakarot_system(&sequencer, EOA_WALLET.clone(), funding_amount).await;
+
+        let starknet_deployer_account = deploy_deployer_account(starknet_provider.clone(), &starknet_account).await;
+
+        // Create a Kakarot client
+        let kakarot_client = KakarotClient::new(
+            KakarotRpcConfig::new(
+                Network::JsonRpcProvider(sequencer.url()),
+                kakarot.kakarot_address,
+                kakarot.proxy_class_hash,
+                kakarot.externally_owned_account_class_hash,
+                kakarot.contract_account_class_hash,
+            ),
+            starknet_provider,
+            starknet_deployer_account,
+        );
+
+        let kakarot_contract = KakarotContract::new(
+            kakarot_client.starknet_provider(),
+            kakarot.kakarot_address,
+            kakarot.proxy_class_hash,
+            kakarot.externally_owned_account_class_hash,
+            kakarot.contract_account_class_hash,
+        );
+
+        // Deploy the EVM contracts
+        let mut evm_contracts = HashMap::new();
 
-        match kakarot.deploy_evm_contract(sequencer.url(), &contract_args.name, contract_args.constructor_args).await {
-            Ok((abi, addresses)) => evm_contracts.insert(contract_args.name, Contract { addresses, abi }),
-            Err(err) => panic!("Failed to deploy contract {}: {:?}", contract_args.name, err.to_string()),
+        // Deploy Counter contract
+        let counter =
```

Comment:
@greged93 did we have a way to make this "automatic", as in parse the contracts available and deploy them, without having to manually add them?

if not lgtm

Reply:
Not really, but maybe we could have a const value which is like a tuple of contract name and constructor args? then we could iterate on this and use `deploy_evm_contract`